### PR TITLE
Fix Subfiling VFD IOC assignment bug

### PR DESCRIPTION
**Pull request by jhendersonHDF**

### CODE REVIEW

File: src/H5FDsubfiling/H5subfiling_common.c [hunk 1]
```diff
@@ -1614,6 +1614,10 @@ identify_ioc_ranks(sf_topology_t *app_topology, int rank_stride)
     assert(app_topology->app_layout->layout);
     assert(app_topology->app_layout->node_count > 0);
```
Comment (brtnfld) [inline src/H5FDsubfiling/H5subfiling_common.c:1615]: Should we add an assert for the maxval of node_count? It was out-of-range in the issue.

Comment (jhendersonHDF) [reply src/H5FDsubfiling/H5subfiling_common.c:1615]: Is there a reasonable value we could use as a max for node count?

Comment (brtnfld) [reply src/H5FDsubfiling/H5subfiling_common.c:1615]: I don't think it can be greater than the number of ranks.

Comment (jhendersonHDF) [reply src/H5FDsubfiling/H5subfiling_common.c:1615]: Added

**Review by brtnfld — APPROVED** [review]
thanks!

**Review by derobins — APPROVED** [review]

**Pull request merged**